matcher/service/openreview_interface.py

```diff
@@ -27,15 +22,30 @@ def __init__(self, client, config_note_id, logger=logging.getLogger(__name__)):
         self._demands = None
         self._constraints = None
 
-        self.validate_score_spec()
-
-    def validate_score_spec(self):
+        self._init_config_attributes()
+        self._validate_score_spec()
+
+    def _init_config_attributes(self):
+        try:
+            self.logger.debug('GET invitation id={}'.format(self.config_note.content['assignment_invitation']))
+            self.assignment_invitation = self.client.get_invitation(self.config_note.content['assignment_invitation'])
+            
+            self.logger.debug('GET invitation id={}'.format(self.config_note.content['assignment_invitation']))
+            self.aggregate_score_invitation = openreview.tools.get_invitation(self.client, self.config_note.content['aggregate_score_invitation'])
+            
+            self.num_alternates = int(self.config_note.content['alternates'])
+        
+        except openreview.OpenReviewException as error_handle:
+            self.set_status('Error', error_handle)
+            raise error_handle
```

Comment:
If you are going to raise the exception anyway, it doesn't make sense to catch the exception here. You can catch the errors in the code that is calling to the interface and set_status. 

Also the interface shouldn't set the error status of the matching run. It should be the matching server main code responsible of that.

Reply:
The way this framework is built right now, we initialize the ConfigNoteInterface object in `routes.py` before creating a thread and calling `Matcher` in the thread.
So, the exceptions that occur in the `__init__` of the class `ConfigNoteInterface` need to be caught either inside the class `__init__` itself OR in `routes.py`.

I could remove the exception handing altogether from the the `__init__` and catch the errors in `routes.py`  but then it is difficult to identify the exact error source.

So, I have removed all the exception handing within `openreview_interface.py` now except for the code within `__init__`. The exception handling code in `__init__` catches the error, then casts it into a custom error and re-raises it. This way the exception can be separately identified in the `routes.py` and will still contain all the traceback result of the original error.

This method of re-raising an exception also allows me to set nice and readable error messages which would appear on the assignments list page.

@melisabok Thoughts?